enclave/tls_cert.c

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/sgxtypes.h>
+#include <openenclave/internal/sha.h>
+#include <openenclave/internal/utils.h>
+#include <openenclave/internal/cert.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/print.h>
+// clang-format on
+//
+#include <stdio.h>
+#include "../common/common.h"
+#include "ec.h"
+#include "key.h"
+#include "rsa.h"
+
+// Todo: consider set CN with enclave's MRENCLAVE values
+#define ISSUER_NAME "CN=Open Enclave SDK,O=OESDK TLS,C=UK"
+#define SUBJECT_NAME ISSUER_NAME
+#define DATE_NOT_VALID_BEFORE "20190501000000"
+#define DATE_NOT_VALID_AFTER "20501231235959"
+
+static unsigned char _cert_buf[MAX_CERT_SIZE] = {
```

Comment:
Is this thread-safe?

Reply:
It is NOT! :-(  Good catch. 
Global memory block was replaced with dynamically allocated local memory.